### ISSUE TITLE
Axis not changing when i rotate my wheel

### ISSUE THREAD

**Issue by JohnSK-dev** [ranenbg/Arduino-FFB-gui#13]
Hello, 
I tried to install the Arduino-FFB-gui onto my PC and Arduino-FFB onto my arduino leonardo. The installation onto the arduino went fine, but when i tried to connect it to the GUI app, the sliders dont respond to the movement of my wheel, but the arduino seems to be recognized in the GUI software (I use leonardo_v170.hex on my arduino). Is there anything that i missed? Could you please help me resolve this issue? Aditionally when I open the DXTweak software the arduino seems to change the sliders as expected.

**Comment by ranenbg** [comment]
Hi, I strongly recommend you to use the lattest firmware v240x and GUI 2.6.1 for the optimal results and problem free operation. You can use manual calibration feature to setup your pedals.

I did mention that lattest gui is backwards compatible all the way down to v170 HEX. But to be honest, I'm no longer checking if that's truly the case. There are so many HEX after a few years of developement and issue you're seeing may be due to some incompatibility of new GUI with old firmware. Just keep everything up to date and all should be fine. I'm testing all new HEX before release and there were no problems.

**Comment by JohnSK-dev** [comment]
I tired the v240ax firmware, but the sliders still dont do anything and I cant use the automatic COM port detection, because the GUI crashes whenever I try running it. But that is not a problem, because i create COM_conf.txt manually and the program turns on, but the problem with sliders may be because the GUI program doesnt show me the binding screen / setup. I dont know if this screen is needed or not, but it was said in the manual and i saw it on some photos. I dont know where the problem may be anymore. I am attaching the binding setup photo from the repository so you know what im talking about. 
![Wheel_control_gamecontrols_setup](https://github.com/user-attachments/assets/bb6bacee-0485-43f6-91a8-a0607ba6390b)

**Comment by ranenbg** [comment]
Do you have the pots correctly connected to analog inputs on Arduino? The pink screen is not required. Upload HEX v240x (without option "a"), you can do manual calibration of pedals. Did you update the gui? You can send the screenshot of it.

**Comment by JohnSK-dev** [comment]
I uploaded the HEX v240x. In this screenshot i had my steering wheel rotated 90 degrees and there was no change in the angle inside the GUI.
![image](https://github.com/user-attachments/assets/f750c327-5a4e-45f3-8a88-47b3f552bb32)

**Comment by ranenbg** [comment]
Yeah, it does seem like there is a problem with axis/button mappings in GUI. It needs to be reset, but I don't know how, as this is part of game controls plus library that I'm using. It is probably somewhere in documents or program data, it needs to be deleted somehow.

**Comment by JohnSK-dev** [comment]
Is this the menu you are referring to? If so, how can i open it ?
![image](https://github.com/user-attachments/assets/fadd4e2b-286f-4aee-8f07-83857f3ad8c5)

**Comment by ranenbg** [comment]
Yes, it should open when you delete com_cfg from data folder of gui. If the gui crashes before you can complete the setup, it's a bummer as I can not reproduce those exact conditions on my PC and can't do anything about it.

You can try to run the gui as sorce code from processing, it should give more info from terminal for troubleshooting. Maybe it will help to narrow down where the problem is.

**Comment by JohnSK-dev** [comment]
I did it and now this is where the program stops 
![image](https://github.com/user-attachments/assets/2ba30153-f180-417a-8ca4-6dfe1867b591)

**Comment by ranenbg** [comment]
Ok, I think I can see where the issue is, tnx for sending this to me. 

So in data folder of gui you will find the button/axis bindings inside the "Leonardo wheel v5" file. You need to place it in the folder where it is missing - this folder where terminal is saying, first red line.

I also see that you have vJoy installed, this is bad news but I hope it will not mess things up on your PC.

**Comment by JohnSK-dev** [comment]
OK i did it and now there is some other issue: 
![image](https://github.com/user-attachments/assets/e20b16f0-efe8-4628-9c39-c69c6108e164)

**Comment by ranenbg** [comment]
Great, that's good news. Now, delete com_cfg from data folder and try again to run gui exe. On your PC Arduino Leonardo (stick) is listed at 15th place, find out to which com port it belongs from device manager, ports. That COM should be in com_cfg, if automatic setup of gui fails again for some reason.

**Comment by JohnSK-dev** [comment]
Yeah it keeps failing with the same error and the exe file still doesnt work, Im gonna try giving it the COM port manually

**Comment by JohnSK-dev** [comment]
The same problem as before, it doesnt update the rotation and the pedal sliders dont work also
![image](https://github.com/user-attachments/assets/7fd2f445-bfc9-4df2-aa5a-a1e671a88ae5)

**Comment by ranenbg** [comment]
Oh man, I see where the issue is. Everything was fine with gui all along, we were sidetracked.

The problem was created because you probably had some other code running in that Arduino, which wrote some registers in EEPROM. Those hold calibration values for pedals. I can see that YR command retrives values 65535 which is max out 16bit number. This has to be fixed. Here's what you need to do.

Option 1. press manual cal button in gui and try to move down the top calibration slider for pedal axis. If it shows no slider then you must send the command to arduino manualy.

Option 2. open Arduino IDE, then open serial monitor and set baud rate 57600 and set carriage return as string terminator. Then open RS232 commands info txt to see which command to send. Look for the one that says pedal calibration.

edit: send following commands:
YA 0
YB 1023
YC 0
YD 1023
YE 0
YF 1023
YG 0
YH 1023
A

You can test if all went fine with command YR, it should give you 8 numbers, the ones you just typed in. If all is good, then close Arduino IDE and try with gui again. Would be helpfull if you send screenshot from DXtweak2, I need to see raw axis calibration values. It is wierd that X-axis for encoder was not working, as it is not related with these calibration values, it has its own fixed at +-32k,  with center at 0.

Would be good to also send a screenshot of device manager where it shows at which COM port Arduino Leonardo is assigned and also some other serial devices if there are any. I don't like that gui is selecting vJoy as a device name (COM12), it should be Arduino Leonardo. If Arduino is indeed at COM12, then vJoy is definately messing things up.

**Comment by JohnSK-dev** [comment]
Thank you, the sliders move now, it was a VJoy conflict, when i deleted VJoy, the steering wheel started working as expected. I can confirm that the arduino was at COM 12 and the manual pedal setup worked. But when i center the steering wheel it sets it to 452 degrees, with which the steering wheel icon isnt upright and this axis doesnt seem to go to negative numbers. I would expect it to be like (450 to -450) with 0 as center point.
![image](https://github.com/user-attachments/assets/a4d261ac-7907-46b3-aa4d-ae9b9a9cb5b1)
![image](https://github.com/user-attachments/assets/11b8203b-8ba6-4b40-a8e3-18dadac18049)

**Comment by ranenbg** [comment]
I'm glad that you managed to solve it by uninstalling vJoy, you don't need that at all.

Have a look at github readme for firmware repository (all the way down) on how to solve incorrect X-axis calibration, where it gets stuck at -540deg.

**Comment by JohnSK-dev** [comment]
Hello,
I was wondering if there was any way of playing Forza Horizon 5 with a steering wheel set up this way? Because I already had my wheel set up another way before and i used Forza Emuwheel to make it happen, but this needs Vjoy to function and there is a problem that Vjoy breaks Arduino-FFB-Gui programs inputs. So I dont know what to do now, or if there is any different way of doing this. Any help with this will be appreciated.

**Comment by ranenbg** [comment]
Hi, please in future make a new issue if you have a new question and close existing thread if I helped you to solve the original problem.

Forza Horizon 5 does not supoort direcInput game controlers. It uses newer xInput protocol and the only way to get it to work is to emulate the DI device into an XI one using various 3rd party software like emuwheel.. You can give a try this:
https://github.com/csutorasa/XOutput
